Server/Controllers/groups.js

```diff
@@ -0,0 +1,341 @@
+
+export default {
+
+/**
+   * Update message flags
+   * Route: GET: /group/updateMessageFlag
+   * @param {Object} firebase firebase object
+   * @param {string} userId user Id string
+   * @param {string} groupId groupId string
+   * @param {string} messageId messageId string
+   * @returns {void} no returns
+   */
+  updateMessageFlag(firebase, userId, groupId, messageId) {
+    const userlogin = firebase.auth().currentUser;
+    if (userlogin) {
+      const firebaseDatabase = firebase.database();
+      const isReadRef = firebaseDatabase
+              .ref(`groups/${groupId}/messages/${messageId}/isRead/${userId}`);
+      isReadRef.once('value', (snap) => {
+        if (snap.val() === null) {
+          firebaseDatabase
+                  .ref(`users/${userId}`)
+                  .once('value', (emailSnap) => {
+                    firebaseDatabase
+                    .ref(`groups/${groupId}/messages/${messageId}/isRead`)
+                    .child(userId).set({
+                      email: emailSnap.val().email
+                    });
+                  });
+        }
+      });
+      const userRef = firebaseDatabase
+                .ref(`users/${userId}/groups/${groupId}/messages/${messageId}`);
+      userRef.once('value', (checkSnap) => {
+        if (checkSnap.val() !== null) {
+          userRef.update({
+            isRead: true
+          });
+        }
+      });
+      const groupRef = firebaseDatabase
+           .ref(`users/${userId}/groups/${groupId}/`);
+      groupRef.update({
+        newMessage: false
+      });
+    }
+  },
+
+  /**
+   * Create new group
+   * Route: POST: /group
+   * @param {Object} req request object
+   * @param {Object} res response object
+   * @param {firebase} firebase firebase object
+   * @returns {Response} response object
+   */
+  create(req, res, firebase) {
+    const userlogin = firebase.auth().currentUser;
+    if (userlogin) {
+      const { groupName, createdBy, dateCreated, createdByUserId, createdByDisplayName } = req.body;
+      const firebaseDatabase = firebase.database();
+      const newKey = firebaseDatabase.ref('groups/').push({
+        groupName,
+        createdBy,
+        dateCreated
+      }).key;
+      const groupRef = firebaseDatabase.ref(`groups/${newKey}/users/`);
+      groupRef.child(createdByUserId).set({
+        userId: createdByUserId,
+        email: createdBy,
+        userName: createdByDisplayName
+      })
+       .then(() => {
+         const userRef = firebaseDatabase
+         .ref(`users/${createdByUserId}/groups/`);
+         userRef.child(newKey).set({
+           groupId: newKey,
+           groupName,
+           newMessage: false
+         });
+         res.send({
+           message: 'New group successfully created',
+         });
+       })
+       .catch((error) => {
+         res.status(500).send({
+           message: `Error occurred ${error.message}`,
+         });
+       });
+    } else {
+      res.status(403).send({
```

Comment:
If a user has to be logged in to access this endpoint but is not, should it be a 403 or a 401?